game-session/src/lib.rs

```diff
@@ -1,22 +1,155 @@
 #![no_std]
-use gstd::{exec, msg, prelude::*, ActorId};
+
+use game_session_io::*;
+use gstd::{exec, msg};
+
+const TRIES_LIMIT: u8 = 5;
+
+static mut GAME_SESSION_STATE: Option<GameSession> = None;
 
 #[no_mangle]
 extern "C" fn init() {
-
+    let game_session_init: GameSessionInit =
+        msg::load().expect("Unable to decode `GameSessionInit`");
+    game_session_init.assert_valid();
+    unsafe { GAME_SESSION_STATE = Some(game_session_init.into()) };
 }
 
 #[no_mangle]
 extern "C" fn handle() {
+    let game_session_action: GameSessionAction =
+        msg::load().expect("Unable to decode `GameSessionAction`");
+    let game_session = unsafe {
+        GAME_SESSION_STATE
+            .as_mut()
+            .expect("Game is not initialized")
+    };
+    match game_session_action {
+        GameSessionAction::StartGame => {
+            let user = msg::source();
+            let session_info = game_session.sessions.entry(user).or_default();
+            match &session_info.session_status {
+                SessionStatus::ReplyReceived(wordle_event) => {
+                    msg::reply::<GameSessionEvent>(wordle_event.into(), 0)
+                        .expect("Error in sending a reply");
+                    session_info.session_status = SessionStatus::WaitUserInput;
+                }
+                SessionStatus::Init
+                | SessionStatus::GameOver(..)
+                | SessionStatus::WaitWordleStartReply => {
+                    let send_to_wordle_msg_id = msg::send(
+                        game_session.wordle_program_id,
+                        WordleAction::StartGame { user },
+                        0,
+                    )
+                    .expect("Error in sending a message");
 
+                    session_info.session_id = msg::id();
+                    session_info.original_msg_id = msg::id();
+                    session_info.send_to_wordle_msg_id = send_to_wordle_msg_id;
+                    session_info.tries = 0;
+                    session_info.session_status = SessionStatus::WaitWordleStartReply;
+                    msg::send_delayed(
+                        exec::program_id(),
+                        GameSessionAction::CheckGameStatus {
+                            user,
+                            session_id: msg::id(),
+                        },
+                        0,
+                        200,
+                    )
+                    .expect("Error in send_delayed a message");
```

Comment:
In the course, there weren't detailed explanations about gas and its significance in real applications. However, you might find it interesting to know that when sending a delayed message, it's better to specify the gas. This gas will be attached to the delayed message and used for processing it.

```suggestion
                    msg::send_with_gas_delayed(
                        exec::program_id(),
                        GameSessionAction::CheckGameStatus {
                            user,
                            session_id: msg::id(),
                        },
                        5_000_000_000,
                        0,
                        200,
                    )
                    .expect("Error in send_delayed a message");
```